reflection.md

```diff
@@ -1,12 +1,15 @@
 # Reflection on implementing regular expressions of a DSL
 
 ## Which operators were easiest to implement and why?
+Implementing regular expressions using a DSL involves defining a set of operators that can be used to construct regular language expressions in a more natural and intuitive way. In this implementation, the easiest operators to implement were the basic operators such as concatenation, union and the Kleene star, which have well-defined and widely accepted semantics. These operators are also relatively straightforward to implement using pattern matching, recursion and other standard programming techniques.
 
 ## Which operators were most difficult to implement and why?
+On the other hand, the most difficult operators to implement were the apply method and the <+> operator. The apply method is used to construct a regular language expression that consists of n repetitions of a given language. This requires recursion and careful handling of the base cases, as well as the use of concatenation to construct the final expression. The <+> operator, which constructs a regular language expression that consists of one or more repetitions of a given language, is also difficult to implement, since it involves constructing a regular expression that matches at least one occurrence of the language followed by any number of additional occurrences.
```

Comment:
I agree that apply was one of the more difficult functions to write; the base case requires careful consideration which may not be immediately obvious.

Reply:
Me Too! Although I ran into some errors aswell. Still learning scala lol